### PR TITLE
REF: dont use numexpr in places where it doesnt help.

### PR DESCRIPTION
**Pull request by jbrockmendel**
DataFrame ops will still use it when dispatching to series/block ops.

Perf neutral:

```
In [3]: arr = np.arange(10**4).reshape(100, 100)                                                                                                                                    
In [4]: df = pd.DataFrame(arr)                                                                                                                                                      
In [5]: df[4] = df[4].astype(float)                                                                                                                                                 
In [6]: ser = df[0]                     
                                                                                                                                            
In [8]: %timeit res = df+ser                                                                                                                                                        
19.8 ms ± 317 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  # <-- PR
19.8 ms ± 267 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  # <-- master

In [10]: %timeit res2 = df.add(ser, axis=0)                                                                                                                                         
19.2 ms ± 276 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  # <-- PR
19.2 ms ± 301 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  # <-- master

In [13]: %timeit res3 = df + df                                                                                                                                                     
23.6 ms ± 382 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  # <-- PR
24 ms ± 551 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  # <-- master
```

### CODE REVIEW

File: pandas/tests/test_expressions.py [hunk 1]
```diff
@@ -95,7 +95,10 @@ def run_binary(self, df, other):
                 expr.get_test_result()
                 result = op(df, other)
                 used_numexpr = expr.get_test_result()
-                assert used_numexpr, "Did not use numexpr as expected."
+
+                # We don't currently use numexpr for comparisons in Series,
```
Comment (jreback) [inline pandas/tests/test_expressions.py]: hmm, the original and entire purpose IS to use numexpr for comparisions. I agree consistency is important, but this is turned off for DataFrames? (for regular dtypes)????

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/test_expressions.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/test_expressions.py]: yah, the right place to do that is in array_ops (which will cause is to do it for Series too)

I've got a branch that implements dataframe-with-dataframe ops block-wise, but the diff for that is pretty big, so im splitting pieces off.  the numexpr-for-series-comparisons part will be next

Comment (jreback) [reply pandas/tests/test_expressions.py]: ok great

yeah comparison ops should. r a pretty big speedup as it’s parallle
computation

Comment (jbrockmendel) [reply pandas/tests/test_expressions.py]: ok, im in the process of splitting off the relevant parts of that branch.  That may end up affecting the tests here, but the core.ops edits here will be correct regardless.

I think we need to improve the used_numexpr check in the tests to not just check whether numexpr was called, but whether it actually was used effectively

Comment (jbrockmendel) [reply pandas/tests/test_expressions.py]: also xref #31297 will smooth this out

**Comment by jbrockmendel** [comment]
closing in favor of #32047.

**Pull request closed without merging**